tasks/fileReplaceTask.go

```diff
@@ -0,0 +1,400 @@
+	return t.Require
+}
+
+func (t *FileReplaceTask) GetPath() string {
+	return t.Path
+}
+
+func (t *FileReplaceTask) String() string {
+	return fmt.Sprintf("task '%s' at path '%s'", t.TypeName, t.GetPath())
+}
+
+func (t *FileReplaceTask) GetOnlyIfCmds() []string {
+	return t.OnlyIf
+}
+
+func (t *FileReplaceTask) GetUnlessCmds() []string {
+	return t.Unless
+}
+
+func (t *FileReplaceTask) GetCreatesFilesList() []string {
+	return t.Creates
+}
+
+func (t *FileReplaceTask) Validate() error {
+	errs := &utils.Errors{}
+
+	err := ValidateRequired(t.Name, t.Path+"."+NameField)
+	errs.Add(err)
+
+	err = ValidateRequired(t.Pattern, t.Path+"."+PatternField)
+	errs.Add(err)
+
+	if len(errs.Errs) > 0 {
+		return errs.ToError()
+	}
+
+	if t.Pattern != "" {
+		compiledRegExp, err := regexp.Compile(t.Pattern)
+		if err != nil {
+			errs.Add(err)
+		}
+		t.PatternCompiled = compiledRegExp
+	}
+
+	if t.MaxFileSize == "" {
+		t.MaxFileSize = defaultMaxFileSize
+	}
+
+	MaxFileSizeCalculated, err := conv.ConvertToFileSize(t.MaxFileSize)
+	if err != nil {
+		errs.Add(err)
+	}
+	t.MaxFileSizeCalculated = MaxFileSizeCalculated
+
+	if t.AppendIfNotFound && t.PrependIfNotFound {
+		errs.Add(ErrAppendAndPrependSetAtTheSameTime)
+	}
+
+	return errs.ToError()
+}
+
+type FileReplaceTaskExecutor struct {
+	FsManager FsManager
+	Runner    tacoexec.Runner
+}
+
+func (frte *FileReplaceTaskExecutor) Execute(ctx context.Context, task Task) ExecutionResult {
+	logrus.Debugf("will trigger '%s' task", task.GetPath())
+	execRes := ExecutionResult{
+		Changes: make(map[string]string),
+	}
+
+	frt, ok := task.(*FileReplaceTask)
+	if !ok {
+		execRes.Err = fmt.Errorf("cannot convert task '%v' to FileReplaceTask", task)
+		return execRes
+	}
+
+	execRes.Name = frt.Name
+
+	var stdoutBuf, stderrBuf bytes.Buffer
+	execCtx := &tacoexec.Context{
+		Ctx:          ctx,
+		Path:         frt.Path,
+		StdoutWriter: &stdoutBuf,
+		StderrWriter: &stderrBuf,
+		Shell:        frt.Shell,
+	}
+
+	logrus.Debugf("will check if the task '%s' should be executed", task.GetPath())
+	skipReason, err := shouldCheckConditionals(execCtx, frte.FsManager, frte.Runner, frt)
+	if err != nil {
+		execRes.Err = err
+		return execRes
+	}
+
+	if skipReason != "" {
+		logrus.Debugf("the task '%s' will be be skipped", task.GetPath())
+		execRes.IsSkipped = true
+		execRes.SkipReason = skipReason
+		return execRes
+	}
+
+	origFilename := frt.Name
+
+	origfileInfo, err := frte.FsManager.Stat(origFilename)
+	if err != nil {
+		execRes.Err = err
+		return execRes
+	}
+
+	if !origfileInfo.Mode().IsRegular() {
+		execRes.Err = fmt.Errorf("%s is not a regular file", origFilename)
+		return execRes
+	}
+
+	if uint64(origfileInfo.Size()) > frt.MaxFileSizeCalculated {
+		logrus.Debugf("the task '%s' will be be skipped", task.GetPath())
+		execRes.IsSkipped = true
+		execRes.SkipReason = "file size is greater than max_file_size"
+		return execRes
+	}
+
+	start := time.Now()
+
+	rollback := false
+	backupFilename := ""
```

Comment:
i've removed the roll back. i think leaving it in has the possibility to be one of those unhelpful "helpful" features that causes problems. forr now at least i think better just to leave the backup file there after an error.

Reply:
I still think this variable should be defined where it's actually used instead of this early in the function. Here:
https://github.com/cloudradar-monitoring/tacoscript/pull/13/files#diff-f853808b1b8f8b21300b70f732214a0c65b26418c84fcb26ea90c14359098222R330

Reply:
yes, understood. but i actually just removed the feature, because trying to restore/rollback when the system is already in an unexpected state can actually create more problems than it solves. i think better just to leave the file there and let the user resolve.